### ISSUE TITLE
Is there interest to use semantic scan code?

### ISSUE THREAD

**Issue by stephanemagnenat** [rust-windowing/winit#1904]
I am working on a GUI library in Rust, aiming at making it backend-independent. So, my library defines abstract events, to be filled by the backend-specific data. For the scan code, I found the [`scancode`](https://github.com/rsaarelm/scancode-rs) crate, based on [libSDL2](https://www.libsdl.org/), itself based on the [USB HID standard](https://www.usb.org/sites/default/files/documents/hut1_12v2.pdf). This seems a good way to encode scan codes, but unfortunately the `scancode` crate seems unmaintained, and I saw that `winit` currently [uses a `u32` for the scan code](https://github.com/rust-windowing/winit/blob/ba704c4eb49638da7516250d26adee35ee0bc9a4/src/event.rs#L725).

So I'm wondering whether there is interest to:
- use semantic scancode matching the USB HID standard, possibly based on the `scancode` crate,
- move this and the `VirtualKeyCode` struct into their own crate as part of `rust-windowing`.

**Comment by maroider** [comment]
>     * use semantic scancode matching the USB HID standard, possibly based on the `scancode` crate,

We're already working on implementing something like this (see #1806 and `src/event.rs` and `src/keyboard.rs` in #1788).

Do note that USB HID scancodes don't get forwarded to applications, as far as I'm aware.

> move this and the `VirtualKeyCode` struct into their own crate as part of `rust-windowing`.

I have had thoughts along these lines myself, and we did consider using `keyboard-types` in Winit while discussing things in #753, but we decided to use our own enumerations and structs for the time being. There was also an attempt to pull out other types into a separate crate (#1298), but that effort has largely stalled.

I have been intending to bring this up again, but I'm currently trying to focus on implementing the API in #753/#1788.
